packages/phoenix-event-display/src/ui/phoenix-menu/phoenix-menu-node.ts

```diff
@@ -214,7 +214,10 @@ export class PhoenixMenuNode {
       return this;
     } else {
       for (const child of this.children) {
-        return child.findInTree(name);
+        const nodeFound = child.findInTree(name);
+        if (nodeFound) {
+          return child.findInTree(name);
```

Comment:
Nice. I almost forgot about this.

So here, we are going through the tree again even when we have the node in `nodeFound` variable.

So just replace this line `return child.findInTree(name);` with `return nodeFound;`.